platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java

```diff
@@ -90,15 +96,77 @@ public static String addressBookConfigText(@NonNull final AddressBook addressBoo
     public static AddressBook parseAddressBookText(@NonNull final String addressBookText) throws ParseException {
         Objects.requireNonNull(addressBookText, "The addressBookText must not be null.");
         final AddressBook addressBook = new AddressBook();
-        for (final String addressLine : addressBookText.split("\\r?\\n")) {
-            final Address address = parseAddressText(addressLine);
-            if (address != null) {
-                addressBook.add(address);
+        boolean nextNodeIdParsed = false;
+        for (final String line : addressBookText.split("\\r?\\n")) {
+            final String trimmedLine = line.trim();
+            if (trimmedLine.isEmpty() || trimmedLine.startsWith("#")) {
+                continue;
+            }
+            if (trimmedLine.startsWith(ADDRESS_KEYWORD)) {
+                final Address address = parseAddressText(trimmedLine);
+                if (address != null) {
+                    addressBook.add(address);
+                }
+            } else if (trimmedLine.startsWith(NEXT_NODE_ID_KEYWORD)) {
+                final NodeId nodeId = parseNextAvailableNodeId(trimmedLine);
+                addressBook.setNextNodeId(nodeId);
+                nextNodeIdParsed = true;
+            } else {
+                throw new ParseException(
+                        "The line [%s] does not start with `address` or `nextAvailableNodeId`."
+                                .formatted(line.substring(0, 30)),
```

Comment:
Might avoid future mismatch to use the constants here:
```suggestion
                        "The line [%s] does not start with `%s` or `%s`."
                                .formatted(line.substring(0, 30), ADDRESS_KEYWORD, NEXT_NODE_ID_KEYWORD),
```